alibi/explainers/similarity/grad.py

```diff
@@ -0,0 +1,203 @@
+"""Gradient-based explainer.
+
+This module implements the gradient-based explainers grad-dot and grad-cos.
+"""
+
+import copy
+from typing import TYPE_CHECKING, Callable, Optional, Union, Dict, Tuple
+
+import numpy as np
+
+from alibi.api.interfaces import Explanation
+from alibi.explainers.similarity.base import BaseSimilarityExplainer
+from alibi.explainers.similarity.metrics import dot, cos, asym_dot
+from alibi.api.defaults import DEFAULT_META_SIM, DEFAULT_DATA_SIM
+
+if TYPE_CHECKING:
+    import tensorflow
+    import torch
+
+
+class SimilarityExplainer(BaseSimilarityExplainer):
+    def __init__(self,
+                 predictor: 'Union[tensorflow.keras.Model, torch.nn.Module]',
+                 loss_fn: '''Callable[[Union[tensorflow.Tensor, torch.Tensor],
+                                    Union[tensorflow.Tensor, torch.Tensor]],
+                                   Union[tensorflow.Tensor, torch.Tensor]]''',
+                 sim_fn: 'Union[Callable[[np.ndarray, np.ndarray], np.ndarray], str]' = 'grad_dot',
+                 task: str = "classification",
+                 store_grads: bool = False,
+                 seed: int = 0,
+                 backend: str = "tensorflow",
+                 **kwargs
+                 ):
+        """Constructor
```

Comment:
The docstring should be descriptive and describe the method, even if only a few sentences. Furthermore, we could add references to a "Notes" section and refer to them via restructuredText hyperlinks from the constructor.